File: core/fitness-dimension/routines/routine/routine.go
package routine

import (
	valuesObjects "fitness-dimension/routines/routine/values-objects"
)

type Training struct {
	name        valuesObjects.RoutineName
	userID      valuesObjects.RoutineUserID
	trainingsID []valuesObjects.RoutineTrainingID
	description valuesObjects.RoutineDescription
}

File: client/src/app/fitness/models/routine.ts
import { Training } from "./training";

export interface Routine {
  id: string
  userId: string
  name: string
  description: string,
  trainings: string[]
}

export type FullRoutine = {
  [key in keyof Routine]: key extends "trainings" ? Training[] : Routine[key]
}
